cmd/oras/internal/option/remote.go

```diff
@@ -273,6 +281,12 @@ func (opts *Remote) NewRepository(reference string, common Common) (repo *remote
 	if repo.Client, err = opts.authClient(hostname, common.Debug); err != nil {
 		return nil, err
 	}
+	logger = logger.WithField("hostname", hostname)
+	repo.HandleWarning = func(warning remote.Warning) {
+		if _, loaded := opts.warned.LoadOrStore(warning.WarningValue, true); !loaded {
+			logger.Warn(warning.Text)
+		}
+	}
```

Comment:
It seems we can refactor this code so that it can also be applied to `NewRegistry()`.
```go
func (opts *Remote) handleWarning(registry string, logger logrus.FieldLogger) func(warning remote.Warning) {
	logger = logger.WithField("registry", registry )
	return func(warning remote.Warning) {
		if _, loaded := opts.warned.LoadOrStore(warning.WarningValue, true); !loaded {
			logger.Warn(warning.Text)
		}
	}
}
```
and
```suggestion
	repo.HandleWarning = opts.handleWarning(hostname, logger)
```

Reply:
With `warned` removed from the `Remote` struct, we don't need to make so many changes for passing the options from values to pointers.

Reply:
Added registry to the key for depulication.

Reply:
@shizhMSFT  If I am copying from `my.registry/repo1` to `my.registry/repo2` and `my.registry` will always give only one warning value, how many warning should be printed out? 
 1 => the map should be in the remote option
 2 => the map can be created within the handler function

Reply:
I see. If they are two repos from the same registry, only 1 should be printed out.

Reply:
will update `sync.Map` to a pointer in remote option

Reply:
updated